PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx

```diff
@@ -680,109 +680,146 @@ struct HfCandidateCreatorXic0Omegac0 {
       KFV0.GetMass(massLam, sigLam);
       if (TMath::Abs(massLam - MassLambda0) > lambdaMassWindow)
         continue;
-      registry.fill(HIST("hKFParticleV0Chi2"), KFV0.GetChi2());
-      if (kfUseV0MassConstraint) {
-        KFV0.SetNonlinearMassConstraint(o2::constants::physics::MassLambda);
-      }
-
+      // err_mass>0 of Lambda
+      if (sigLam <= 0)
+        continue;
+      // chi2>0 && NDF>0 for selecting Lambda
+      if ((KFV0.GetNDF() <= 0 || KFV0.GetChi2() <= 0))
+        continue;
+      kfOmegac0Candidate.chi2GeoV0 = KFV0.GetChi2();
       KFParticle KFV0_m = KFV0;
       KFV0_m.SetNonlinearMassConstraint(o2::constants::physics::MassLambda);
-
-      //-------------------------- V0 info---------------------------
-      // pseudorapidity
-      float pseudorapV0Dau0 = trackV0Dau0.eta();
-      float pseudorapV0Dau1 = trackV0Dau1.eta();
-
-      // info from from KFParticle
-      std::array<float, 3> pVecV0 = {KFV0.GetPx(), KFV0.GetPy(), KFV0.GetPz()}; // pVec stands for vector containing the 3-momentum components
-      std::array<float, 3> vertexV0 = {KFV0.GetX(), KFV0.GetY(), KFV0.GetZ()};
-      std::array<float, 3> pVecV0Dau0 = {kfpPos.GetPx(), kfpPos.GetPy(), kfpPos.GetPz()};
-      std::array<float, 3> pVecV0Dau1 = {kfpNeg.GetPx(), kfpNeg.GetPy(), kfpNeg.GetPz()};
+      if (kfUseV0MassConstraint) {
+        KFParticle KFV0 = KFV0_m;
+      }
 
       //__________________________________________
-      //*>~<* step 2 : reconstruc cascade(Omega) with KF
-      KFParticle kfpV0 = KFV0;
-      const KFParticle* OmegaDaugthers[2] = {&kfpBachKaon, &kfpV0};
+      //*>~<* step 2 : reconstruct cascade(Omega) with KF
+      const KFParticle* OmegaDaugthers[2] = {&kfpBachKaon, &KFV0};
       // construct cascade
       KFParticle KFOmega;
       KFOmega.SetConstructMethod(kfConstructMethod);
       try {
         KFOmega.Construct(OmegaDaugthers, 2);
       } catch (std::runtime_error& e) {
-        LOG(debug) << "Failed to construct omega from V0 and bachelor track: " << e.what();
+        LOG(debug) << "Failed to construct Omega from V0 and bachelor track: " << e.what();
         continue;
       }
       float massCasc, sigCasc;
       KFOmega.GetMass(massCasc, sigCasc);
+      // err_massOmega > 0
+      if (sigCasc <= 0)
+        continue;
+      if (TMath::Abs(massCasc - MassOmegaMinus) > massToleranceCascade)
+        continue;
+      // chi2>0 && NDF>0
+      if (KFOmega.GetNDF() <= 0 || KFOmega.GetChi2() <= 0)
+        continue;
+      kfOmegac0Candidate.chi2GeoCasc = KFOmega.GetChi2();
+      KFParticle KFOmega_m = KFOmega;
+      KFOmega_m.SetNonlinearMassConstraint(o2::constants::physics::MassOmegaMinus);
```

Comment:
What does the suffix `_m` stand for? Please choose a more meaningful name and avoid underscores.

Reply:
It's done. `_m` stands for MassConstrained KFparticle, now change variables name as : KFV0_m -> kfV0MassConstrained, KFOmega_m -> kfOmegaMassConstrained